image/image.go

```diff
@@ -97,24 +97,24 @@ func validate(w walker, refs []string, out *log.Logger) error {
 // UnpackLayout walks through the file tree given by src and, using the layers
 // specified in the manifest pointed to by the given ref, unpacks all layers in
 // the given destination directory or returns an error if the unpacking failed.
-func UnpackLayout(src, dest, ref string) error {
-	return unpack(newPathWalker(src), dest, ref)
+func UnpackLayout(u *Unpacker, src, dest, ref string) error {
```

Comment:
I feel tempted but `UnpackLayout` (through the helper `unpack`) seems to be doing more than the `Unpacker`. Namely, `UnpackLayout` finds the descriptor, validates it, finds the manifest, validates it, and then proceeds to the actuall process of unpacking the manifest.

I feel that the current boundary between `Unpacker` and this logic is healthy (`Unpacker` only knows about a manifest and a walker in the image layout and deals with the process of unpacking the layout). Of course these boundaries usually get shifted as projects move forward but I am comfortable with how things are split now.

Do you have strong feelings about this @wking?